### DISCUSSION TITLE
Model methods that take arrays as arguments should also be allowed to take Collections

### DISCUSSION THREAD

**Discussion by BrekiTomasson** [laravel/framework#43389] [Ideas]
Say I have the following:

```php
>>> $country = Country::firstWhere('name', 'Sweden');
=> App\Models\World\Country {#5391
     id: 197,
     name: "Sweden",
     alpha2: "SE",
     alpha3: "SWE",
     language: null,
     demonym: null,
     pop_millions: null
   }
```

If I wanted to update the `null` fields, I can do so using the instantiated variable or directly on the Model class, but I have to use an array:

```php
>>> $country->update(['language' => 'Swedish', 'demonym' => 'Swede', 'pop_millions' => 10]);

>>> Country::where('id', $country->id)->update(['language' => 'Swedish', 'demonym' => 'Swede', 'pop_millions' => 10]);
```

But I can't do this:

```php
>>> $country = Country::firstWhere('name', 'Sweden');
>>> $changes = new Illuminate\Support\Collection()
>>> $changes->put('demonym', 'Swede')
>>> $changes->put('language', 'Swedish')
>>> $changes->put('pop_millions', 10)
>>> $country->update($changes);
```

I know it's as easy as doing `$country->update($changes->toArray())`, of course, but it's not immediately *obvious* why updating a Model using an "arrayable" collection like this shouldn't be handled natively. Under the hood, it could be as simple as rewriting the methods in `Illuminate\Database\Eloquent` to something like (using `update()` as an example):

```php
    public function update(array|BaseCollection $attributes = [], array $options = [])
    {
        if ($attributes instanceof BaseCollection) {
            $attributes = $attributes->toArray();
        }

        if (! $this->exists) {
            return false;
        }

        return $this->fill($attributes)->save($options);
    }
```

**Comment by michael-rubel** [comment]
Hi @BrekiTomasson 

This one would be a breaking change if was added at the minor release. I'm not sure it wouldn't introduce bugs across the codebase. Will require a code review in a lot of places definitely, so I don't see it earlier than in Laravel `10.x` or even `11.x`.
